index.js

```diff
@@ -42,7 +42,7 @@ function makeRegexp(pattern, options) {
 		return regexpCache.get(cacheKey);
 	}
 
-	const negated = pattern[0] === '!';
+	const negated = pattern[0] === '!' ? 1 : 0;
```

Comment:
This feels like a premature optimization. I would prefer using a boolean.

Reply:
The reason is bitwise xor at index.js:100. But I agree - code clarity probably prevails over miniscule drop of speed.